src/app/getting-started/services/getting-started.service.spec.ts

```diff
@@ -0,0 +1,236 @@
+  ExtUser,
+  GettingStartedService
+} from './getting-started.service';
+
+describe('GettingStartedService', () => {
+
+  type TestContext = {
+    service: GettingStartedService;
+    controller: HttpTestingController;
+  };
+
+  beforeEach(function(this: TestContext): void {
+    TestBed.configureTestingModule({
+      imports: [ HttpClientTestingModule ],
+      providers: [
+        GettingStartedService,
+        {
+          provide: AuthenticationService,
+          useFactory: (): jasmine.SpyObj<AuthenticationService> => {
+            const svc: jasmine.SpyObj<AuthenticationService> = createMock(AuthenticationService);
+            svc.getToken.and.returnValue('mock-auth-token');
+            return svc;
+          }
+        },
+        {
+          provide: UserService,
+          useFactory: (): jasmine.SpyObj<UserService> => {
+            const svc: jasmine.SpyObj<UserService> = createMock(UserService);
+            return svc;
+          }
+        },
+        {
+          provide: Logger,
+          useFactory: (): jasmine.SpyObj<Logger> => {
+            const logger: jasmine.SpyObj<Logger> = createMock(Logger);
+            return logger;
+          }
+        },
+        { provide: WIT_API_URL, useValue: 'https://example.com/api/' }
+      ]
+    });
+    this.service = TestBed.get(GettingStartedService);
+    this.controller = TestBed.get(HttpTestingController);
+  });
+
+  it('should be instantiable', function(this: TestContext): void {
+    expect(this.service).toBeDefined();
+  });
+
+  describe('#createTransientProfile', () => {
+    it('should return transient profile', function(this: TestContext): void {
+      const userData: ExtUser = {
+        attributes: {
+          username: 'example-user',
+          contextInformation: {
+            foo: 'bar'
+          }
+        }
+      } as ExtUser;
+      TestBed.get(UserService).loggedInUser = Observable.of(userData);
+      expect(this.service.createTransientProfile()).toEqual(userData.attributes);
+    });
+
+    it('should populate empty contextInformation', function(this: TestContext): void {
+      const userData: ExtUser = {
+        attributes: {
+          username: 'example-user'
+        }
+      } as ExtUser;
+      TestBed.get(UserService).loggedInUser = Observable.of(userData);
+      expect(this.service.createTransientProfile()).toEqual({
+        username: 'example-user',
+        contextInformation: {}
+      } as ExtProfile);
+    });
+
+    it('should return an empty object if no user is logged in', function(this: TestContext): void {
+      TestBed.get(UserService).loggedInUser = Observable.never();
+      expect(this.service.createTransientProfile()).toEqual({} as ExtProfile);
+    });
+  });
+
+  describe('HTTP Tests', () => {
+
+    afterEach(function(this: TestContext): void {
+      this.controller.verify();
+    });
+
+    describe('#getExtProfile', () => {
+      it('should send a GET', function(this: TestContext, done: DoneFn): void {
```

Comment:
Can this be simplified to just `function(done: DoneFn): void {` ?

Reply:
If I extract the references contained in the `TestContext` out and just hold them in an enclosing scope or something, sure, but it's as it is intentionally.

Reply:
Ok. I didn't look closely enough and with the param named `this` it was confusing.